lib/nats.js

```diff
@@ -749,6 +761,9 @@ Client.prototype.sendConnect = function() {
     if (this.options.nkey !== undefined) {
         cs.nkey = this.options.nkey;
     }
+    if (this.options.noEcho) {
+        cs.echo = false;
```

Comment:
Humm... `cs.echo` should be initialized to `true` and set to `false` if `noEcho` option is set (as you do). So without `noEcho: true` your test that check that by default a connection receives its own message should fail?

Reply:
interesting - not sure why the test doesn't fail.

Reply:
perhaps on the server unless the option is returned, it doesn't do anything

Reply:
Regardless I cannot set it unless they provide the option, because if the server doesn't support the option, it will fail.

Reply:
Ok. After discussion, it works because the server defaults the "echo" boolean to true in the structure that is unmarshal'ed when processing the client CONNECT. So that old clients that don't set this work ok (that is with "echo" semantics).
IMO, you should set it to `true` by default and to `false` if user sets NoEcho, but either way works. (I believe that if you set to `true` and connect to a server that does not know about the `echo` field in CONNECT, nothing bad will happen).

Reply:
Seems to work correctly under 1.1.0 - but keeping it as is for right now.